LOG_SIMPLE_MSG_OPTIMIZE breaks qemu CI tests

**Describe the bug**

Starting with commit a68cdb0558a4d369316a24c931ed1eff29f0cda0 some CI tests for qemu_x86 started failing, e.g. here:
https://github.com/zephyrproject-rtos/zephyr/pull/64543/checks?check_run_id=18180911999

I can reproduce the issue locally, and if I force CONFIG_LOG_SIMPLE_MSG_OPTIMIZE=n they start working again.

**To Reproduce**

This doesn't work:
```
west build -p -b qemu_x86 tests/subsys/logging/log_benchmark -T logging.benchmark_user -t run
```

But this works:
```
west build -p -b qemu_x86 tests/subsys/logging/log_benchmark -T logging.benchmark_user -t run -- -DCONFIG_LOG_SIMPLE_MSG_OPTIMIZE=n
```

**Logs and console output**
```
SeaBIOS (version rel-1.16.2-0-gea1b7a073390-prebuilt.qemu.org)
Booting from ROM..
*** Booting Zephyr OS build zephyr-v3.5.0-671-g55c59e041e45 ***
Running TESTSUITE test_log_benchmark
===================================================================
LOGGING MODE:DEFERRED
	OVERWRITE: 1
	BUFFER_SIZE: 2048
	SPEED: 0START - test_log_capacity
In total 471 message were stored.
 PASS - test_log_capacity in 0.008 seconds
===================================================================
START - test_log_message_store_time_no_overwrite
Average logging a message:  1364 cycles (13 us)
 PASS - test_log_message_store_time_no_overwrite in 0.014 seconds
===================================================================
START - test_log_message_store_time_no_overwrite_from_user
FAILED: zephyr/CMakeFiles/run_qemu /Users/jhedberg/src/zephyr/zephyr/build/zephyr/CMakeFiles/run_qemu
cd /Users/jhedberg/src/zephyr/zephyr/build && /usr/local/bin/qemu-system-i386 -m 32 -cpu qemu32,+nx,+pae -machine q35 -device isa-debug-exit,iobase=0xf4,iosize=0x04 -no-reboot -nographic -no-acpi -net none -pidfile qemu.pif
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /usr/local/bin/cmake --build /Volumes/case-sensitive/zephyr/zephyr/build --target run
```

Should this new option really default to `y`, or should I disable it explicitly for the affected platforms?

--- a/include/zephyr/logging/log_msg.h
+++ b/include/zephyr/logging/log_msg.h
@@ -532,7 +532,7 @@ do { \
 				    UTIL_AND(UTIL_NOT(_domain_id), UTIL_NOT(_cstr_cnt))), \
 			( \
 			bool can_simple = LOG_MSG_SIMPLE_CHECK(__VA_ARGS__); \
-			if (can_simple && ((_dlen) == 0)) { \
+			if (can_simple && ((_dlen) == 0) && !k_is_user_context()) { \
 				LOG_MSG_DBG("create fast message\n");\
 				Z_LOG_MSG_SIMPLE_ARGS_CREATE(_domain_id, _source, _level, \
 						     Z_LOG_FMT_ARGS(_fmt, ##__VA_ARGS__)); \

--- a/tests/subsys/logging/log_benchmark/testcase.yaml
+++ b/tests/subsys/logging/log_benchmark/testcase.yaml
@@ -18,7 +18,7 @@ tests:
 
   logging.benchmark_user:
     integration_platforms:
-      - native_posix
+      - qemu_x86
     tags: logging
     platform_allow:
       - qemu_x86

